lib/components/RunFrame.tsx

```diff
@@ -39,30 +39,62 @@ interface Props {
 
 export const RunFrame = (props: Props) => {
   const [circuitJson, setCircuitJson] = useState<any>(null)
+  const [error, setError] = useState<{
+    phase?: string
+    componentDisplayName?: string
+    error?: string
+    stack?: string
+  } | null>(null)
 
   useEffect(() => {
+    let worker: Awaited<ReturnType<typeof createCircuitWebWorker>>
+
     async function runWorker() {
-      const worker = await createCircuitWebWorker({
-        webWorkerUrl: evalWebWorkerBlobUrl,
-        verbose: true,
-      })
-      const $finished = worker.executeWithFsMap({
-        entrypoint: props.entrypoint,
-        fsMap: props.fsMap,
-      })
-      console.log("waiting for execution to finish...")
-      await $finished
-      console.log("waiting for initial circuit json...")
-      setCircuitJson(await worker.getCircuitJson())
-      console.log("got initial circuit json")
-      await $finished.catch((e) => {
+      try {
+        worker = await createCircuitWebWorker({
+          webWorkerUrl: evalWebWorkerBlobUrl,
+          verbose: true,
+        })
```

Comment:
this part doesn't belong in the try catch, try to keep the try catch as small as possible